wasmbinding/stargate_whitelist_test.go

```diff
@@ -0,0 +1,114 @@
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/query"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+
+	"github.com/osmosis-labs/osmosis/v11/app"
+	"github.com/osmosis-labs/osmosis/v11/wasmbinding"
+)
+
+type StargateTestSuite struct {
+	suite.Suite
+
+	ctx sdk.Context
+	app *app.OsmosisApp
+}
+
+func (suite *StargateTestSuite) SetupTest() {
+	suite.app = app.Setup(false)
+	suite.ctx = suite.app.BaseApp.NewContext(false, tmproto.Header{Height: 1, ChainID: "osmosis-1", Time: time.Now().UTC()})
+}
+
+func TestStargateTestSuite(t *testing.T) {
+	suite.Run(t, new(StargateTestSuite))
+}
+
+func (suite *StargateTestSuite) TestDeterministicJsonMarshal() {
+	testCases := []struct {
+		name             string
+		originalResponse string
+		updatedResponse  string
+		queryPath        string
+		expectedProto    proto.Message
+	}{
+
+		/**
+		   * Origin Response
+		   * balances:<denom:"bar" amount:"30" > pagination:<next_key:"foo" >
+		   * "0a090a036261721202333012050a03666f6f"
+		   *
+		   * New Version Response
+		   * The binary built from the proto response with additional field address
+		   * balances:<denom:"bar" amount:"30" > pagination:<next_key:"foo" > address:"cosmos1j6j5tsquq2jlw2af7l3xekyaq7zg4l8jsufu78"
+		   * "0a090a036261721202333012050a03666f6f1a2d636f736d6f73316a366a357473717571326a6c77326166376c3378656b796171377a67346c386a737566753738"
+		   // Origin proto
+		   message QueryAllBalancesResponse {
+		  	// balances is the balances of all the coins.
+		  	repeated cosmos.base.v1beta1.Coin balances = 1
+		  	[(gogoproto.nullable) = false, (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"];
+		  	// pagination defines the pagination in the response.
+		  	cosmos.base.query.v1beta1.PageResponse pagination = 2;
+		  }
+		  // Updated proto
+		  message QueryAllBalancesResponse {
+		  	// balances is the balances of all the coins.
+		  	repeated cosmos.base.v1beta1.Coin balances = 1
+		  	[(gogoproto.nullable) = false, (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"];
+		  	// pagination defines the pagination in the response.
+		  	cosmos.base.query.v1beta1.PageResponse pagination = 2;
+		  	// address is the address to query all balances for.
+		  	string address = 3;
+		  }
+		*/
+		{
+			"Query All Balances",
+			"0a090a036261721202333012050a03666f6f",
+			"0a090a036261721202333012050a03666f6f1a2d636f736d6f73316a366a357473717571326a6c77326166376c3378656b796171377a67346c386a737566753738",
+			"/cosmos.bank.v1beta1.Query/AllBalances",
+			&banktypes.QueryAllBalancesResponse{
+				Balances: sdk.NewCoins(sdk.NewCoin("bar", sdk.NewInt(30))),
+				Pagination: &query.PageResponse{
+					NextKey: []byte("foo"),
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		suite.Run(fmt.Sprintf("Case %s", tc.name), func() {
+			suite.SetupTest()
+
+			binding, ok := wasmbinding.StargateWhitelist.Load(tc.queryPath)
```

Comment:
I think we can have a separate fixture (`TestStargateWhitelistLoad`) just for this where we can validate that all of the queries we expect are loaded correctly + have a test case where a query path fails because it is not whitelisted

Reply:
Sure! I can add a test case for testing cases that hasnt been whitelisted.

Reply:
wdym by 

> where we can validate that all of the queries we expect are loaded correctly

Looks kind of hard to me to validate through all queries by directly calling them

Reply:
What I was thinking is to have a test case for each of the queries we have whitelisted right now:
https://github.com/osmosis-labs/osmosis/blob/204f20ad1ee487bdd723121264ce71783ddf43c6/wasmbinding/stargate_whitelist.go#L20-L23

+ 1 that is not whitelisted and expect that after `SetupTest()`,
```
binding, ok := wasmbinding.StargateWhitelist.Load(tc.queryPath)
```
ok is false

I'm not sure of all the limitations. Is there anything specific blocking us from doing this in a separate new test fixture `TestStargateWhitelistLoad`?

Reply:
For the test case that is not whitelisted, it is tested in stargate queier. 
The reason I think testing the whitelists individually is not a sustainable test case is because we now only have 3 whitelist to test, which is not something hard to hard code, but future plans are to add 50+ whitelists, which would then be meaningless to test individually by hard coding the paths!